File: easy/1-two-sum.cxx
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> diffs;
        for(int i = 0; i != nums.size(); ++i) {
            int diff = target - nums[i];
            if(diffs.find(diff) != diffs.end()) {
                return std::vector<int> {diffs[diff], i};
            }
            diffs[nums[i]] = i;
        }
        return std::vector<int>();
    }
};

File: easy/1-two-sum.rs
use std::collections::HashMap;

struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut diffs: HashMap<i32, i32>= HashMap::new();
        for (i, n) in nums.iter().enumerate() {
            let diff = target - n;
            if diffs.contains_key(&diff) {
                return vec![i as i32, diffs[&diff]];
            }
            diffs.insert(*n, i as i32);
        }
        return vec![];
    }
}

fn main() {

}
